forgerock-openbanking-uk-aspsp-rs/forgerock-openbanking-uk-aspsp-rs-gateway/forgerock-openbanking-uk-aspsp-rs-gateway-server/src/test/java/com/forgerock/openbanking/aspsp/rs/wrappper/endpoints/PaymentsSubmissionEndpointWrapperTest.java

```diff
@@ -0,0 +1,192 @@
+/**
+ * Copyright 2019 ForgeRock AS.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.forgerock.openbanking.aspsp.rs.wrappper.endpoints;
+
+import com.forgerock.openbanking.am.config.AMOpenBankingConfiguration;
+import com.forgerock.openbanking.am.services.AMResourceServerService;
+import com.forgerock.openbanking.aspsp.rs.wrappper.RSEndpointWrapperService;
+import com.forgerock.openbanking.common.conf.RSConfiguration;
+import com.forgerock.openbanking.common.model.openbanking.forgerock.ConsentStatusCode;
+import com.forgerock.openbanking.common.model.openbanking.forgerock.FRPaymentConsent;
+import com.forgerock.openbanking.common.model.openbanking.v3_0.payment.FRDomesticConsent1;
+import com.forgerock.openbanking.common.services.openbanking.OBHeaderCheckerService;
+import com.forgerock.openbanking.constants.OIDCConstants;
+import com.forgerock.openbanking.exceptions.OBErrorException;
+import com.forgerock.openbanking.jwt.services.CryptoApiClient;
+import com.forgerock.openbanking.model.error.ErrorCode;
+import dev.openbanking4.spring.security.multiauth.model.authentication.PasswordLessUserNameAuthentication;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.UUID;
+
+import static com.forgerock.openbanking.integration.test.support.JWT.jws;
+import static org.assertj.core.api.Assertions.*;
+
+
+@RunWith(SpringJUnit4ClassRunner.class)
+public class PaymentsSubmissionEndpointWrapperTest {
+
+    private static PaymentsSubmissionsEndpointWrapper wrapper;
+
+    @MockBean(name = "cryptoApiClient")
```

Comment:
If you're writing a unit test you should use mockito to mock the object

Reply:
I have started using mockito to mock the Wrapper RSEndpointWrapperService but this Service doesn't have default constructor and mockito cannot mock the primitive types, final classes and anonymous classes, the RSEndpointWrapperService constructor primitive types.

Reply:
Why do you need to mock primitive types? final or anonymous classes?